packages/astro/src/core/dev/dev.ts

```diff
@@ -34,6 +40,45 @@ export default async function dev(inlineConfig: AstroInlineConfig): Promise<DevS
 	const restart = await createContainerWithAutomaticRestart({ inlineConfig, fs });
 	const logger = restart.container.logger;
 
+	const currentVersion = process.env.PACKAGE_VERSION ?? '0.0.0';
+	const isPrerelease = currentVersion.includes('-');
+
+	if (!isPrerelease) {
+		try {
+			// Don't await this, we don't want to block the dev server from starting
+			shouldCheckForUpdates(restart.container.settings.preferences).then(async (shouldCheck) => {
```

Comment:
One minor thing I just remembered, could we skip the version check when running tests or in CI? So we don't spam npm everytime we run tests 😅